src/Table.hxx

```diff
@@ -108,6 +108,62 @@ public:
 
     std::vector<std::pair<std::string, std::string>> flatten_properties() const;
 
+    //====================================================================
+
+    inline std::map<size_t, Data_Type> determine_datatypes_for_writing(
+            const Format &format) const {
+        std::map<size_t, Data_Type> alt_datatype_lookup;
+        bool adjust_uint64 = ((format.enum_format == Format::Enums::VOTABLE) ||
+                              (format.enum_format == Format::Enums::IPAC_TABLE) ||
+                              (format.enum_format == Format::Enums::FITS));
+        bool adjust_all_unsigned = (format.enum_format == Format::Enums::VOTABLE);
```

Comment:
So, I just want to point out that the FITS standard does in fact support unsigned integers in a roundabout way. See section 5.2.5 of [the standard](https://fits.gsfc.nasa.gov/standard40/fits_standard40aa-le.pdf) - the gist is that you set the (binary table) column type to be a signed m-bit integer (where m = 16, 32, or 64), and then specify an offset by setting TSCALn  = 1 and TZEROn = 2^(m - 1). You have to subtract 2^(m - 1) from unsigned values before storage to get appropriate signed values.

In other words, gymnastics are required for all unsigned integer types in FITS. But! As far as I can tell, CFITSIO handles the pain for you. Or at least the version I am looking at (3.45) `#defines` TUINT, TUSHORT, TULONGLONG, so I don't think you need to do any type conversions at a higher level.

Reply:
Ugh. The version of CFITSIO in Debian Jessie doesn't have TULONGLONG; upgrading to Buster will fix that. So I guess we should keep the special handling for FITS, but add comments indicating the FITS workaround should be removed as soon as we have a sufficiently recent version of CFITSIO to play with.

Reply:
Will do.